shhext/service.go

```diff
@@ -141,15 +144,21 @@ func (t *tracker) handleEvent(event whisper.EnvelopeEvent) {
 		if !ok || state == EnvelopeSent {
 			return
 		}
+		log.Debug("envelope is sent", "hash", event.Hash, "peer", event.Peer)
+		t.cache[event.Hash] = EnvelopeSent
 		if t.handler != nil {
-			log.Debug("envelope is sent", "hash", event.Hash, "peer", event.Peer)
-			t.handler(event.Hash)
-			t.cache[event.Hash] = EnvelopeSent
```

Comment:
Good catch :)

Reply:
why catch?
i changed it mainly for a cleaner tracker tests

Reply:
Because previously, if `handler` was not defined, `t.cache` was never updated here, but it was used in `case whisper.EventEnvelopeExpired:`.

Reply:
it didn't matter much because i deleted a hash only if it was in the cache